manifests/init.pp

```diff
@@ -260,7 +269,29 @@
     validate_absolute_path($keys_real)
   }
 
-  validate_string($restrict_options)
+  if is_array($restrict_options) == true {
+    $restrict_options_real = $restrict_options
+  }
+  # needed for backward compatibility
+  elsif is_string($restrict_options) == true {
+    $restrict_options_real = $restrict_options ? {
+      'USE_DEFAULTS' => $default_restrict_options,
+      default        => [ "-4 $restrict_options", "-6 $restrict_options", ]
+    }
+  }
+  else {
+    fail("restrict_options must be an array (prefered) or a string (will be deprecated).")
```

Comment:
*It's a suggestion only*
Well, the new handling in the template is using an arrays. In my eyes it is preferable to have input and output variables in the same format. It avoids possible conversation errors and surprises on the user side. If you use a string it result in two lines with ```-4``` and ```-6``` as prefix. Using an array it will output exaxtly your input. 

Above string to array conversation is a dirty workaround in my eyes. It's (hopefully) only there to not break things and expectations until the next major version update.